### PR TITLE
DietPi-Software | PiVPN: Add unattended install

### LINKED ISSUE #3350: DietPi-Software | PiVPN: Make install unattended?
The PiVPN installer allows unattended install via given inputs. We could either use this always or depending on interactive/non-interactive call (automated firstrun installs). Needs some testing to find working+reasonable defaults.

### PR DESCRIPTION
**Pull request by bastianpaetzold**
This is a possible implementation of unattended install of PiVPN (#3350)

I know there are probably still open points like if  `/boot/unattended_pivpn.conf` should be renamed or deleted after the installation, but I hope this can serve as a starting point. Please let me know if I should adjust anything.

I tested the installation with and without the `/boot/unattended_pivpn.conf` present as well as reinstall after an unattended install.

### CODE REVIEW

File: dietpi/dietpi-software [hunk 1]
```diff
@@ -8117,12 +8118,20 @@ _EOF_
 			G_DIETPI-NOTIFY 2 'Preventing install of unattended-upgrades'
 			G_EXEC sed -i 's/^[[:blank:]]*UNATTUPG=[[:digit:]]/UNATTUPG=0/' install.bash
 
+			local install_options
+
+			if [[ -f /boot/unattended_pivpn.conf ]]; then
+				G_EXEC sed -i 's/^[[:blank:]]*UNATTUPG=[[:digit:]]/UNATTUPG=0/' /boot/unattended_pivpn.conf
+				install_options='--unattended /boot/unattended_pivpn.conf'
```
Comment (MichaIng) [inline dietpi/dietpi-software]:
```suggestion
				install_options=('--unattended' '/boot/unattended_pivpn.conf')
```

File: dietpi/dietpi-software [hunk 2]
```diff
@@ -8117,12 +8118,20 @@ _EOF_
...
+			fi
+
 			# APT deps
 			G_AGI dnsutils net-tools bsdmainutils iptables-persistent # https://github.com/pivpn/pivpn/blob/master/auto_install/install.sh#L37
 			Configure_iptables
 
-			G_EXEC_OUTPUT=1 G_EXEC_NOEXIT=1 G_EXEC ./install.bash || aSOFTWARE_INSTALL_STATE[$software_id]=0
+			G_EXEC_OUTPUT=1 G_EXEC_NOEXIT=1 G_EXEC ./install.bash $install_options || aSOFTWARE_INSTALL_STATE[$software_id]=0
```
Comment (MichaIng) [inline dietpi/dietpi-software]: Shellcheck doesn't like it when non-explicit variable splitting is done, hence we use arrays in such cases:
```suggestion
			G_EXEC_OUTPUT=1 G_EXEC_NOEXIT=1 G_EXEC ./install.bash "${install_options[@]}" || aSOFTWARE_INSTALL_STATE[$software_id]=0
```

File: dietpi/dietpi-software [hunk 3]
```diff
@@ -8117,12 +8118,20 @@ _EOF_
...
 			G_EXEC_NOHALT=1 G_EXEC rm install.bash
+			[[ -f /boot/unattended_pivpn.conf ]] && G_EXEC_NOHALT=1 G_EXEC mv /boot/unattended_pivpn.conf{,.ignore}
```
Comment (MichaIng) [inline dietpi/dietpi-software]: To be more clear?
```suggestion
			[[ -f /boot/unattended_pivpn.conf ]] && G_EXEC_NOHALT=1 G_EXEC mv /boot/unattended_pivpn.conf{,.applied}
```

File: dietpi/dietpi-software [hunk 4]
```diff
@@ -8117,12 +8118,20 @@ _EOF_
 			G_DIETPI-NOTIFY 2 'Preventing install of unattended-upgrades'
 			G_EXEC sed -i 's/^[[:blank:]]*UNATTUPG=[[:digit:]]/UNATTUPG=0/' install.bash
 
+			local install_options
+
+			if [[ -f /boot/unattended_pivpn.conf ]]; then
+				G_EXEC sed -i 's/^[[:blank:]]*UNATTUPG=[[:digit:]]/UNATTUPG=0/' /boot/unattended_pivpn.conf
```
Comment (MichaIng) [inline dietpi/dietpi-software]: What is this setting used for? Generally, if users set it to something other than `0`, shouldn't we respect that? Generally, we have some nice function to apply settings transparently 🙂:
```suggestion
				G_CONFIG_INJECT 'UNATTUPG=[[:digit:]]' 'UNATTUPG=0' /boot/unattended_pivpn.conf
```

File: dietpi/dietpi-software (hunk 3 above)
Comment (bastianpaetzold) [reply dietpi/dietpi-software]: Yes, way better

File: dietpi/dietpi-software (hunk 4 above)
Comment (bastianpaetzold) [reply dietpi/dietpi-software]: I totally agree and I normally wouldn't add something like this, but we already have this for interactive install: `G_EXEC sed -i 's/^[[:blank:]]*UNATTUPG=[[:digit:]]/UNATTUPG=0/' install.bash` causes `UNATTUPG=0` to be set regardless of the users input (if I understand it correctly). 
So I would prefer to respect the users choice in either case but didn't want to have inconsistent behavior between interactive and unattended install.

Comment (MichaIng) [reply dietpi/dietpi-software]: Ah, it's about the `unattended-upgrades` package, not about PiVPN install/upgrades being unattended. Yes then it makes sense 👍. You can use the inject function above. The unattended upgrades question can be actually skipped more effectively. I'll add this as well.

**Comment by MichaIng** [comment]
### Testing
- [ ] Install without `unattended_pivpn.conf`
- [ ] with `unattended_pivpn.conf`

**Review by MichaIng — DISMISSED** [review]

**Review by MichaIng — DISMISSED** [review]

**Comment by Joulinar** [comment]
should we give an example or hint on our online docs how the `/boot/unattended_pivpn.conf` could/should looks like.

**Comment by MichaIng** [comment]
Issue opened: https://github.com/MichaIng/DietPi-Docs/issues/736

**Review by MichaIng — APPROVED** [review]

**Pull request merged**